### PR TITLE
Registry Event Hubs conn string provider only when conn string is not empty

### PR DESCRIPTION
**Pull request by moarychan**
# Description

Fixes https://github.com/Azure/azure-sdk-for-java/issues/42880
Registry Event Hubs conn string provider only when conn string is not empty

If using connection string:

```yaml
spring:
  cloud:
    azure:
      eventhubs:
        connection-string: ${AZURE_EVENTHUBS_CONNECTION_STRING}
    function:
      definition: consume;supply
    stream:
      bindings:
        consume-in-0:
          destination: demo
          group: $Default
        supply-out-0:
          destination: demo
      binders:
        kafka:
          environment:
            spring:
              main:
                sources: com.azure.spring.cloud.autoconfigure.implementation.eventhubs.kafka.AzureEventHubsKafkaAutoConfiguration

```

then you can add environment variables to use other credentials:
```shell
spring.cloud.azure.eventhubs.connection-string=
spring.cloud.stream.binders.kafka.environment.spring.main.sources=
spring.cloud.stream.kafka.binder.brokers=<event-hubs-namespace-name>.servicebus.windows.net:9093
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Release History
 
+## 5.19.0-beta.1 (Unreleased)
+
+### Spring Cloud Azure Autoconfigure
+This section includes changes in `spring-cloud-azure-autoconfigure` module.
+
+#### Bugs Fixed
+- Fix to registry Event Hubs connection string provider only when conn string is not empty. [#42919](https://github.com/Azure/azure-sdk-for-java/issues/42880).
```
Comment (rujche) [inline sdk/spring/CHANGELOG.md:9]: `Fix to registry Event Hubs connection string provider only when conn string is not empty.` -> `Fix error: Event Hubs connection string is still used when it's configured to empty string`

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java [hunk 2]
```diff
@@ -45,7 +48,8 @@ AzureEventHubsProperties azureEventHubsProperties() {
     }
 
     @Bean
-    @ConditionalOnProperty("spring.cloud.azure.eventhubs.connection-string")
+    @ConditionalOnExpression("'${spring.cloud.azure.eventhubs.connection-string:}' != ''")
+    @ConditionalOnMissingBean(value = AzureServiceType.EventHubs.class, parameterizedContainer = ServiceConnectionStringProvider.class)
```
Comment (rujche) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java:52]: Why is this necessary?

Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java:52]: Not required, but the same support of conn string availability is also useful for Event Hubs configuration.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/kafka/AzureEventHubsKafkaAutoConfigurationTests.java [hunk 3]
```diff
@@ -38,118 +38,127 @@ class AzureEventHubsKafkaAutoConfigurationTests {
     @Test
     void shouldNotConfigureWhenAzureEventHubsKafkaDisabled() {
         this.contextRunner
-                .withPropertyValues("spring.cloud.azure.eventhubs.kafka.enabled=false")
-                .run(context -> assertThat(context).doesNotHaveBean(AzureEventHubsKafkaAutoConfiguration.class));
+            .withPropertyValues("spring.cloud.azure.eventhubs.kafka.enabled=false")
+            .run(context -> assertThat(context).doesNotHaveBean(AzureEventHubsKafkaAutoConfiguration.class));
```
[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/kafka/AzureEventHubsKafkaAutoConfigurationTests.java:42]: The original indentation was incorrect.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java (hunk 2 above)
Comment (rujche) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java:52]: > Not required

Then how about deleting it?

> but the same support of conn string availability is also useful for Event Hubs configuration.

I don't understand this.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/kafka/AzureEventHubsKafkaAutoConfigurationTests.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (rujche) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/kafka/AzureEventHubsKafkaAutoConfigurationTests.java:42]: OK.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java (hunk 2 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java:52]: Communicated offline

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java:52]: I think it is because we are injecting a `StaticConnectionStringProvider` bean here. @moarychan, but why we are using conditional on missing bean here?

File: sdk/spring/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
 # Release History
 
+## 5.19.0-beta.1 (Unreleased)
+
+### Spring Cloud Azure Autoconfigure
+This section includes changes in `spring-cloud-azure-autoconfigure` module.
+
+#### Features Added
+- Registry a connection string provider for Event Hubs only when conn string is not empty; support to custom the connection string provider for Event Hubs. [#42919](https://github.com/Azure/azure-sdk-for-java/pull/42919).
```
Comment (rujche) [inline sdk/spring/CHANGELOG.md:9]: `Registry a connection string provider for Event Hubs only when conn string is not empty; support to custom the connection string provider` -> `Support customize StaticConnectionStringProvider<AzureServiceType.EventHubs> bean`.

**Review by rujche — APPROVED** [review]
LGTM.
